docs/cl/libvirt.md

```diff
@@ -0,0 +1,245 @@
+
+## Requirements
+
+* At least 2 GiB free ram and 20 GiB free disk
+* NetworkManager in dnsmasq mode
+* libvirt / virsh
+* Terraform v0.11.x and 
+    - [terraform-provider-libvirt ](https://github.com/dmacvicar/terraform-provider-libvirt)
+    - [terraform-provider-ct ](https://github.com/coreos/terraform-provider-ct)
+
+## Libvirt
+
+Libvirt is a suite of tools that manages virtual machines, storage, and networking. It does not run virtual machines directly, rather it relies on a lower-level virtualization engine such as qemu or bhyve.
+
+You will need the `virsh` and `qemu-img` binaries installed on your system:
+
+```
+sudo dnf install libvirt-client qemu-img
+```
+
+## CoreOS Base image
+
+You will need to manually download a CoreOS base image. Qemu supports a copy-on-write format, so this will allow for deduplication.
+
+```
+wget https://stable.release.core-os.net/amd64-usr/current/coreos_production_qemu_image.img.bz2
+bunzip2 coreos_production_qemu_image.img.bz2
+qemu-img resize coreos_production_qemu_image.img +8G
+```
+
+Make a note of the absolute path to this image, you'll need it later.
+
+
+## DNS
+
+Libvirt will start a DNS server for each cluster, and will create a DNS name for every node. However, we need to make the autogenerated names resolvable on the local machine. In order to do this, you will need to put NetworkManager in to `dnsmasq` mode, which will route all DNS queries through a local dnsmasq instance. Then we can instruct dnsmasq to delegate queries for a specific domain to the libvirt resolver.
+
+You will need to select values for your `domain` and, if overridden, `node_ip_pool` Terraform variables below. Use them here.
+
+For example, if you use a `domain` of `hestia.k8s` and the default  `node_ip_pool` of `192.168.120.0/24`, then your DNS server will be at `192.168.120.1`
+
+1. Edit `/etc/NetworkManager/NetworkManager.conf` and set `dns=dnsmasq` in section `[main]`
+2. Tell dnsmasq to use your cluster. The syntax is `server=/<domain>/<firstIP>`. For this example:
+```
+echo server=/hestia.k8s/192.168.120.1 | sudo tee /etc/NetworkManager/dnsmasq.d/typhoon.conf
+```
+3. `systemctl restart NetworkManager`
+
+## Terraform Setup
+
+Install [Terraform](https://www.terraform.io/downloads.html) v0.11.x on your system.
+
+```sh
+$ terraform version
+Terraform v0.11.1
+```
+
+Install [terraform-provider-libvirt ](https://github.com/dmacvicar/terraform-provider-libvirt).
+
+```sh
+ go get github.com/dmacvicar/terraform-provider-libvirt
+mkdir -p ~/.terraform.d/plugins
+cp $GOPATH/bin/terraform-provider-libvirt ~/.terraform.d/plugins/
+```
+
+Install [terraform-provider-ct ](https://github.com/coreos/terraform-provider-ct).
+```sh
+ go get github.com/coreos/terraform-provider-ct
+mkdir -p ~/.terraform.d/plugins
+cp $GOPATH/bin/terraform-provider-ct ~/.terraform.d/plugins/
```

Comment:
Match the others. https://github.com/poseidon/typhoon/pull/254#issuecomment-401570522